File: common/src/main/java/com/zorroa/common/elastic/AbstractElasticDao.java
package com.zorroa.common.elastic;

import com.zorroa.common.config.ApplicationProperties;
import org.elasticsearch.client.Client;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;

public abstract class AbstractElasticDao {

    protected final Logger logger = LoggerFactory.getLogger(getClass());

    protected Client client;
    protected ElasticTemplate elastic;

    @Autowired
    public void setApplicationProperties(ApplicationProperties props) {

    }

    @Autowired
    public void setClient(Client client) {
        this.client = client;
        this.elastic = new ElasticTemplate(client, getIndex(), getType());
    }

    public abstract String getType();
    public abstract String getIndex();

    public void refreshIndex() {
        client.admin().indices().prepareRefresh(getIndex()).get();
    }
}

File: archivist/src/main/kotlin/com/zorroa/archivist/elastic/AbstractElasticDao.kt
package com.zorroa.archivist.elastic

import com.zorroa.archivist.config.ApplicationProperties
import com.zorroa.archivist.security.getOrgId
import com.zorroa.archivist.service.IndexRoutingService
import com.zorroa.common.clients.EsRestClient
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired

abstract class AbstractElasticDao {

    protected val logger = LoggerFactory.getLogger(javaClass)

    protected lateinit var indexRoutingService: IndexRoutingService
    protected lateinit var elastic: ElasticTemplate

    @Autowired
    fun setApplicationProperties(props: ApplicationProperties) {

    }

    @Autowired
    fun setup(routingService: IndexRoutingService) {
        this.indexRoutingService = routingService
        this.elastic = ElasticTemplate(routingService)
    }

    fun refreshIndex() { }

    fun getClient() : EsRestClient {
        return indexRoutingService[getOrgId()]
    }
}
